imageio/plugins/tifffile.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, imageio contributors
+# imageio is distributed under the terms of the (new) BSD License.
+
+""" Storage of image data in tiff format.
+"""
+
+from __future__ import absolute_import, print_function, division
+
+from .. import formats
+from ..core import Format
+
+import numpy as np
+
+tifffile = None  # Defer loading to lib() function.
+
+
+def lib():
+    global tifffile
+    try:
+        import tifffile
+    except ImportError:
+        from . import _tifffile as tifffile
+    return tifffile
+
+
+TIFF_FORMATS = ('.tif', '.tiff', '.stk', '.lsm')
+WRITE_METADATA_KEYS = ('photometric', 'planarconfig', 'resolution',
+                       'description', 'compress', 'volume', 'writeshape',
+                       'extratags')
+READ_METADATA_KEYS = ('planar_configuration', 'is_fluoview', 'is_nih',
+                      'is_contiguous', 'is_micromanager', 'is_ome',
+                      'is_palette', 'is_reduced', 'is_rgb', 'is_sgi',
+                      'is_shaped', 'is_stk', 'is_tiled',
+                      'resolution_unit', 'compression',
+                      'fill_order', 'orientation', 'predictor')
```

Comment:
How about `Metadata on reading` and `Metadata for Writing` sections in the class docstring?